website/public/js/controllers/tasksCtrl.js

```diff
@@ -129,6 +132,24 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
 
     /*
      ------------------------
+     Dailies
+     ------------------------
+     */
+    $scope.updateTaskStartDate = function(task) {
+      // TODO: Weirdness here...startDate should be a Date and _dateString
+      // should be a string, but the datePicker input fields sets _dateString
+      // to a Date even though input type="text"...
+      task.startDate = task._dateString;
+    };
+
+    $scope.open = function($event) {
```

Comment:
Couple of things
1. Let's name this something more specific. `$scope.openDatePicker` maybe?
2. Let's pass in the task as a second argument and set value for whether or not the datepicker should be open on the task itself. So:
```
$scope.openDatePicker = function($event, task) {
    $event.stopPropagation();
    task.datePickerOpen = !task.datePickerOpen
}
```

Then in the view, have the input appear on `task.datePickerOpen` instead.

The way you have it now, if multiple tasks are open in edit mode, the datepicker will appear for all of them if it's called on any one of them.